client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -303,23 +326,15 @@ public void flush() {
     
     private void flushInternal() {
         boolean success = false;
-        String sealedSegment = null;
         while (!success) {
             success = true;
```

Comment:
Updated the code to handle this case. Used exceptions as a method to convey the fact that a segment sealed was encountered during the flush. It ensures that the data sent to the new segment is also flushed.